### PR TITLE
Feature/if updatable true only authed can vote 201

### PR DESCRIPTION
**Pull request by colorimeter**
TC:
- cannot_cast_a_user_if_canUpdate_is_true_but_not_logged_in

### CODE REVIEW

File: src/main/java/org/rulez/demokracia/pdengine/VoteRegistry.java [hunk 1]
```diff
@@ -7,6 +7,8 @@
 import org.json.JSONObject;
 import org.rulez.demokracia.pdengine.exception.ReportedException;
 
+import com.fasterxml.jackson.databind.Module.SetupContext;
```
Comment (magwas) [inline src/main/java/org/rulez/demokracia/pdengine/VoteRegistry.java]: Why?

File: src/main/java/org/rulez/demokracia/pdengine/VoteRegistry.java [hunk 2]
```diff
@@ -19,11 +21,17 @@ public String obtainBallot(String id, String adminKey) {
 
 		if (adminKey.equals(vote.adminKey))
 			vote.increaseRecordedBallots("admin");
-		else if (adminKey.equals("user"))
+
+		else if (adminKey.equals("user")) {
+			if (getWsContext().getUserPrincipal() == null)
+				throw new IllegalArgumentException("Anon admin is not authenticated, cannot issue any ballot.");
```
Comment (magwas) [inline src/main/java/org/rulez/demokracia/pdengine/VoteRegistry.java]: "The user is not authenticated, cannot issue a ballot without the admin key" would be a better eror message.

File: src/test/java/org/rulez/demokracia/pdengine/ObtainBallotAdminKeyTest.java [hunk 3]
```diff
@@ -55,4 +55,37 @@ public void if_neededAssurances_is_empty_then_a_ballot_is_served_to_anyone() {
...
+	@tested_behaviour("if adminkey is anon, only one ballot can be issued")
+	@Test
+	public void even_if_the_user_does_have_all_the_assurances_he_cannot_issue_more_than_one_ballot() {
+		Vote vote = voteManager.getVote(adminInfo.voteId);
+		vote.neededAssurances.clear();
+		vote.neededAssurances.add("magyar");
+		
+		String ballot = voteManager.obtainBallot(adminInfo.voteId, "user");
+		assertNotEquals("", ballot );
+		System.out.println("Princi:" + (voteManager.getWsContext().getUserPrincipal()== null));
```
Comment (magwas) [inline src/test/java/org/rulez/demokracia/pdengine/ObtainBallotAdminKeyTest.java]: please do not leave printlns behind.

**Review by magwas — CHANGES_REQUESTED** [review]

File: src/main/java/org/rulez/demokracia/pdengine/VoteRegistry.java (hunk 1 above)
Comment (colorimeter) [reply src/main/java/org/rulez/demokracia/pdengine/VoteRegistry.java]: dunno, eclipse mistery

File: src/main/java/org/rulez/demokracia/pdengine/VoteRegistry.java (hunk 2 above)
Comment (colorimeter) [reply src/main/java/org/rulez/demokracia/pdengine/VoteRegistry.java]: ok

File: src/test/java/org/rulez/demokracia/pdengine/ObtainBallotAdminKeyTest.java (hunk 3 above)
Comment (colorimeter) [reply src/test/java/org/rulez/demokracia/pdengine/ObtainBallotAdminKeyTest.java]: what a shame, ok.

**Review by magwas — APPROVED** [review]
nice work
testcasecount=1

**Pull request merged**